src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs

```diff
@@ -75,5 +78,11 @@ public CompilerCacheResult(IList<IChangeToken> expirationTokens)
         /// Gets a value that determines if the view was successfully found and compiled.
         /// </summary>
         public bool Success { get; }
+
+        /// <summary>
+        /// Gets a Func to create the view.
+        /// </summary>
+        /// <remarks>This property is not available when <see cref="Success"/> is <c>false</c>.</remarks>
+        public Func<object> PageCreate { get; }
```

Comment:
@rynowak  and @pranavkm  I don't see any value of having the CompilationResult in the CompilerCacheResult once I create the PageCreator delegate. Is it safe to remove that property from the struct?